rand_core/src/impls.rs

```diff
@@ -58,28 +58,21 @@ macro_rules! fill_via_chunks {
         let chunk_size_u8 = min($src.len() * SIZE, $dst.len());
         let chunk_size = (chunk_size_u8 + SIZE - 1) / SIZE;
 
-        // The following can be replaced with safe code, but unfortunately it's
-        // ca. 8% slower.
-        if cfg!(target_endian = "little") {
-            unsafe {
-                core::ptr::copy_nonoverlapping(
-                    $src.as_ptr() as *const u8,
-                    $dst.as_mut_ptr(),
-                    chunk_size_u8);
-            }
-        } else {
-            for (&n, chunk) in $src.iter().zip($dst.chunks_mut(SIZE)) {
-                let tmp = n.to_le();
-                let src_ptr = &tmp as *const $ty as *const u8;
-                unsafe {
-                    core::ptr::copy_nonoverlapping(
-                        src_ptr,
-                        chunk.as_mut_ptr(),
-                        chunk.len());
-                }
+        // Byte-swap for portability of results:
+        if cfg!(target_endian = "big") {
+            for x in &mut $src[..chunk_size] {
+                *x = x.to_le();
             }
         }
 
+        // We do a simple copy, which is 25-50% faster:
```

Comment:
Faster than what? This comment made sense when it was LE vs BE, but now that this code is run unconditionally, the comment doesn't make sense anymore.